admin-cli/src/index.ts

```diff
@@ -100,42 +110,137 @@ namespace v3v4 {
+      }
+    }
   }
 
-  const [file] = process.argv.slice(3);
-  const json = await fs.readFile(file, {encoding: 'utf8'});
-  const config = JSON.parse(json) as Config;
+  // eslint-disable-next-line no-inner-declarations
+  export function parse (): Cli {
+    const argv: Arguments = require('yargs')
+      .usage('Usage $0 <command> [options]')
+      .version(false)
+      .help('h')
+      .alias('h', 'help')
+      .command({
+        command: 'v3-init',
+        desc: 'Initialze a test ledger',
+        builder: (yargs: Argv) => {
+          yargs.option('f', {
+            alias: 'file',
+            demandOption: true,
+            describe: 'Configuration file',
+            type: 'string',
+          })
+        },
+      })
+      .example('$0 v3-init -f test-setup.json', 'Ledger initialization from configuration file.')
+      .command({
+        command: 'v3-v4-upgrade',
+        desc: 'Create v3/v4 upgrade proposals',
+        builder: (yargs: Argv) => {
+          yargs.option('f', {
+            alias: 'file',
+            demandOption: true,
+            describe: 'Configuration file',
+            type: 'string',
+          })
+        }
+      })
+      .example('$0 v3-v4-upgrade test-setup.json', 'Creation of v3/v4 upgrade proposal contracts')
+      .command({
+        command: 'v3-v4-upgrade-accept',
+        desc: 'Exercise accepts on v3/v4 upgrade proposals.',
+        builder: (yargs: Argv) => {
+          yargs
+            .option('ledger-url', { demandOption: true, describe: 'Ledger URL', type: 'string' })
+            .option('ledger-id', { demandOption: true, describe: 'Ledger ID', type: 'string' })
+            .option('application-id', { demandOption: true, describe: 'Application ID', type: 'string' })
+            .option('secret', { demandOption: true, describe: 'Secret', type: 'string' })
+        }
+      })
+      .example('$0 v3-v4-upgrade-accept --ledger-url ...', 'Exercising accept choices on v3/v4 upgrade proposals')
+      .demandCommand(1, 'Missing command')
+      .epilogue('$0 is an administration tool - use with caution!')
+      .argv;
 
-  switch (process.argv[2]) {
-    case 'v3-init': {
-      await v3.init(config);
-      break;
-    }
-    case 'v3-v4-upgrade': {
-      await v3v4.upgrade(config);
-      break;
-    }
-    case 'v3-v4-upgrade-accept': {
-      await v3v4.accept(config);
-      break;
-    }
-    default: {
-      console.log ('Unrecognized command "' + process.argv[2] + '"');
-      usage();
-      break;
+    return argsToCli(argv);
+  }
+
+  // eslint-disable-next-line no-inner-declarations
+  async function config(file: string): Promise<Config> {
+    const json = await fs.readFile(file, { encoding: 'utf8' });
+    return JSON.parse(json) as Config;
+  }
+
+  // eslint-disable-next-line no-inner-declarations
+  export async function run (cli: Cli): Promise<void> {
+    switch (cli.command) {
+      case 'v3-init': {
+        await v3.init(await config(cli.file));
```

Comment:
```suggestion
        const config = await readConfig(cli.file);
        await v3.init(config);
```
This makes the order of evaluation, namely which `await` happens first, easier to understand.

Reply:
Will fix. Follow up PR!